src/sentry/tasks/base.py

```diff
@@ -78,3 +78,21 @@ def wrapped(*args, **kwargs):
         return wrapped
 
     return inner
+
+
+def track_group_async_operation(function):
+    def wrapper(*args, **kwargs):
+        try:
+            response = function(*args, **kwargs)
+        except Exception:
+            metrics.incr("group.update.async_task", sample_rate=1.0, tags={"status": 500})
+            # Continue raising the error now that we've incr the metric
+            raise
+        metrics.incr(
+            "group.update.async_task",
+            sample_rate=1.0,
+            tags={"status": 200 if response is True else 500},
```

Comment:
I'm worried using `response is True` is brittle since it would be easy to overlook that this decorator means that the response has to be True on success.

Are there cases where a task has finished but it wasn't successful?

Reply:
I was worried about it too but wasn't sure of a better way. Maybe not using a decorator...or maybe something that helps prevent this requirement from being overlooked.

I don't think there is a place where this happens...there are a couple of places in `sync_status_outbound` that are questionable - not sure if the feature flag or issue being missed should result in a 500 or not.

Reply:
Also to confirm a suspicion of my own, upstream status has to match http codes to be treated properly?

and hmmm I think if an error wasn't thrown you could treat that as a success, and then use `return False` to mark spots as explicit failures instead?

Reply:
I switched the decorator's logic `500 if response is False else 200` and modified the functions to return False as failures - removing the need to return True